wsmaster/che-core-sql-schema/src/main/resources/che-schema/6.10.0/2__change_signature_key_pair_id.sql

```diff
@@ -0,0 +1,36 @@
+--
+-- Copyright (c) 2012-2018 Red Hat, Inc.
+-- This program and the accompanying materials are made
+-- available under the terms of the Eclipse Public License 2.0
+-- which is available at https://www.eclipse.org/legal/epl-2.0/
+--
+-- SPDX-License-Identifier: EPL-2.0
+--
+-- Contributors:
+--   Red Hat, Inc. - initial API and implementation
+--
+
+-- Rename old table
+ALTER TABLE che_sign_key_pair RENAME TO che_sign_key_pair_old;
+
+-- Create new key pair table
+CREATE TABLE che_sign_key_pair (
+    workspace_id        VARCHAR(255) NOT NULL,
+    public_key          BIGINT       NOT NULL,
+    private_key         BIGINT       NOT NULL,
+
+    PRIMARY KEY (workspace_id)
+);
+-- Constraint
+ALTER TABLE che_sign_key_pair ADD CONSTRAINT fk_sign_workspace_id FOREIGN KEY (workspace_id) REFERENCES workspace (id);
+-- Index  (old indexes still preserved)
+CREATE INDEX index_sign_private_key_pair_id ON che_sign_key_pair (workspace_id);
+
+-- Copy data
+INSERT  INTO che_sign_key_pair
+SELECT r.workspace_id, k.public_key, k.private_key
+FROM che_k8s_runtime AS r,
+(SELECT public_key, private_key from che_sign_key_pair_old LIMIT 1) as k;
+
+-- Cleanup
+DROP TABLE che_sign_key_pair_old CASCADE;
```

Comment:
new line

Reply:
> You mean why cascade?

Yes. I agree that it is safer like migration won't fail, but if something related to records that are migrated then maybe it would be better to migrate these records too.

So, `CASCADE` here may prevent migration failing but it also can leads to unexpected removing of data.

Reply:
It's interesting question how this related data will behave after renaming the original table. Is the references still be valid?

Reply:
Must be not related to name. But anyway i prefer to leave cascade because we know that there is no related data (anf it it were exists we will need to tie it to the new table manulally anyway), an it makes migration more stable.